src/APIBasedAuth.ts

```diff
@@ -0,0 +1,192 @@
+import axios, {
+  AxiosError,
+  AxiosInstance,
+  AxiosRequestConfig,
+  AxiosResponse
+} from 'axios';
+
+/* istanbul ignore next */
+const formatAxiosError = <ErrorResponse = any>(
+  error: AxiosError<ErrorResponse> | unknown
+) => {
+  if (!axios.isAxiosError(error)) {
+    throw new AxiosError(String(error));
+  }
+  throw error.response?.data;
+};
+
+class APIBasedAuth {
+  private client: AxiosInstance;
+  private clientOptions: APIBasedAuth.Config;
+  private session?: APIBasedAuth.Session;
+  private tokenData?: APIBasedAuth.TokenData;
+
+  constructor(options: APIBasedAuth.Config) {
+    const required = ['baseURL', 'clientId'];
```

Comment:
Can we make `baseURL` optional, and if it is not defined, assume the production url: `https://apps.us.lifeomic.com/auth/v1/api`